docs/developer/web.rst

```diff
@@ -0,0 +1,224 @@
+or collections of resources. The plural form should be used as endpoint names.
+Whenever possible, a single word should be used to describe resources. Where
+compound words cannot be avoided, they are built using **dashes**. For complex
+queries (filtering, sorting) query parameters **should** be used.
+
+
+- URL for the entire collection of contexts:
+    ``/contexts``
+
+- URL for a specific context:
+    ``/contexts/{id}``
+
+
+5.1 Relations
+~~~~~~~~~~~~~
+A relation between resources **must** be represented in the URL structure if
+the related resources only exist in the context of the relation. For example,
+if in a situation where a file has primary key composed of a bucket identifier
+and a file identifier, the URLs would be structured as followed:
+
+- URL for all files inside the bucket:
+    ``/buckets/{bucket}/files``
+
+- URL for a specific file in a given bucket:
+    ``/buckets/{bucket}/files/{file}``
+
+If a related resource does exist independently of the related resource, relations
+are still represented in the URL structure, however individual resources are
+accessed independently.
+
+- URL for all executions related to a given context:
+    ``/contexts/{id}/executions``
+
+All relations of a resource **should** be made available following such patterns.
+
+- ``/projects/{id}/creator`` - single resource can be a ``301`` redirect to the
+  resource base URL (e.g. ``Location: /users/1234``).
+- ``/projects/{id}/buckets``
+
+In contrast, a specific resource is accessed independently of the relations which
+have been followed.
+
+- URL for a specific execution whose id was retrieved trough a context:
+    ``/executions/{id}``
+
+To make following relations more convenient, links to the related resources **should**
+be included in the response body when a given relation is queried. In case of a single
+resource response the links **may** also be included in the response `link header
+<https://tools.ietf.org/html/rfc5988#page-6>`_.
+
+
+6. Query Parameters
+-------------------
+
+6.1 Filtering
+~~~~~~~~~~~~~
+
+Unique query parameters **must** be used for each field that allows filtering.
+
+- Retrieving all executions which are currently running:
+    ``GET /executions?status=running``
+
+
+6.2 Sorting
+~~~~~~~~~~~
+A generic ``sort`` parameter **must** be used to describe sorting rules. If there
+are multiple fields that a collection can be sorted by, the ``sort`` parameter
+**must** accept a list of comma separated fields, indicating descending sort
+priority. A minus sign implies descending sort order in the subsequent field.
+
+- Retrieving all files, ordered alphabetically by file type and by descending file size:
+    ``GET /files?sort=type,-size``
+
+6.3 Pagination
+~~~~~~~~~~~~~~
+
+Pagination **must** be supported by all endpoints returning more than just a single
+resource. Pagination **must be** implemented using the query parameters ``page``
+and ``per_page``.
```

Comment:
maybe "page_size" ? (I've seen it more often. Especially while working with django-rest-framework)

Reply:
This can be an open discussion + voting.